packages/web-frontend/src/sagas.js

```diff
@@ -924,6 +924,13 @@ function* watchTryUpdateMeasureConfigAndWaitForHierarchyLoad() {
   );
 }
 
+function* updateMapOverlayOnceHierarchyLoads() {
+  yield take(FETCH_MEASURES_SUCCESS);
+  const state = yield select();
+  const currentMapOverlayId = selectCurrentMapOverlayId(state);
+  yield put(setMapOverlay(currentMapOverlayId));
+}
+
```

Comment:
Previously when we refresh or hit the URL with map overlay id, we fetch measure data before the map overlay hierarchy is not loaded, which cannot include the period in the request and shows all the data. Now we always fetch measure data after hierarchy is loaded.